### PR TITLE
finish buttons for toolbar

### PR DESCRIPTION
**Pull request by clewis7**
finishes #149 

added record button, need to make image widget toolbar

### CODE REVIEW

File: fastplotlib/layouts/_gridplot.py [hunk 1]
```diff
@@ -399,3 +405,12 @@ def update_current_subplot(self, ev):
                 self.panzoom_controller_button.value = subplot.controller.enabled
                 self.maintain_aspect_button.value = subplot.camera.maintain_aspect
 
+    def record_plot(self, obj):
+        if self.record_button.value:
+            try:
+                self.plot.record_start(f"./{datetime.now().isoformat()}.mp4")
```
Comment (kushalkolar) [inline fastplotlib/layouts/_gridplot.py:411]: ISO format returns colons and periods, so replace those with underscores. I think windows will complain with those characters. I think you can just do `.replace()` on a str

File: fastplotlib/layouts/_gridplot.py [hunk 2]
```diff
@@ -399,3 +405,12 @@ def update_current_subplot(self, ev):
...
+            except ModuleNotFoundError or FileExistsError:
```
Comment (kushalkolar) [inline fastplotlib/layouts/_gridplot.py:412]: just allow broad `Exception` if a weirder issue occurs

File: fastplotlib/widgets/image.py [hunk 3]
```diff
@@ -856,7 +865,161 @@ def show(self):
...
+        if toolbar and self.toolbar is None:
+            self.toolbar = ImageWidgetToolbar(self).widget
+            return VBox([canvas, self.toolbar, self.widget])
+        elif toolbar and self.toolbar is not None:
+            return VBox([canvas, self.toolbar, self.widget])
+        else:
+            return VBox([canvas, self.widget])
+
+
+class ImageWidgetToolbar:
```
Comment (kushalkolar) [inline fastplotlib/widgets/image.py:883]: most of the stuff in here is not necessary, you can just get the toolbar from the `Plot` or `GridPlot` instance.

File: fastplotlib/widgets/image.py [hunk 4]
```diff
@@ -856,7 +865,161 @@ def show(self):
...
+        if self.record_button.value:
+            try:
+                self.plot.record_start(f"./{datetime.now().isoformat()}.mp4")
+            except ModuleNotFoundError or FileExistsError:
+                traceback.print_exc()
+                self.record_button.value = False
+        else:
+            self.plot.record_stop()
+
+    def link_play(self):
```
Comment (kushalkolar) [inline fastplotlib/widgets/image.py:1012]: I don't this this method is necessary, just call `jlink` in __init__?

**Review by kushalkolar — CHANGES_REQUESTED** [review]

**Comment by kushalkolar** [comment]
what about this for flip: https://fontawesome.com/icons/arrows-up-down?f=sharp&s=light

**Comment by clewis7** [comment]
got rid of redundant code, `iw.show()` method might need some work but everything is functional

File: fastplotlib/widgets/image.py [hunk 5]
```diff
@@ -590,7 +599,7 @@ def __init__(
 
         # TODO: So just stack everything vertically for now
         self.widget = VBox([
-            self.plot.canvas,
+            # self.plot.canvas,
```
Comment (kushalkolar) [inline fastplotlib/widgets/image.py:602]:
```suggestion
```

**Review by kushalkolar — APPROVED** [review]

**Pull request merged**